src/Propulsion.Cosmos/PropulsionCosmosPrometheus.fs

```diff
@@ -4,111 +4,145 @@ namespace Propulsion.CosmosStore.Prometheus
+
+    let private objectives =
+        [|
+            0.50, 0.05 // Between 45th and 55th percentile
+            0.95, 0.01 // Between 94th and 96th percentile
+            0.99, 0.01 // Between 100th and 98th percentile
+        |] |> Array.map Prometheus.QuantileEpsilonPair
+
+    let private create' statSuffix descSuffix (tagNames, tagValues) stat desc =
+        let config =
+            Prometheus.SummaryConfiguration(Objectives = objectives, LabelNames = append tagNames groupLabels, MaxAge = System.TimeSpan.FromMinutes 1.)
+
+        create config (baseName stat + statSuffix) (baseDesc desc + descSuffix) tagValues
+
+    
+    let latency = create' secondsStat latencyDesc
+    let charge = create' rusStat chargeDesc
 
 module private Histogram =
 
-    let private create (cfg : Prometheus.HistogramConfiguration) name desc =
-        let h = Prometheus.Metrics.CreateHistogram(name, desc, cfg)
-        fun app (db, con, group) v -> h.WithLabels(app, db, con, group).Observe(v)
-    // Given we also have summary metrics with equivalent labels, we focus the bucketing on LAN latencies
-    let private sHistogram =
-        let sBuckets = [| 0.0005; 0.001; 0.002; 0.004; 0.008; 0.016; 0.5; 1.; 2.; 4.; 8.; 16. |]
-        let sCfg = Prometheus.HistogramConfiguration(Buckets = sBuckets, LabelNames = Impl.groupLabels)
-        create sCfg
-    let private ruHistogram =
-        let ruBuckets = Prometheus.Histogram.ExponentialBuckets(1., 2., 9) // 1 .. 256
-        let ruCfg = Prometheus.HistogramConfiguration(Buckets = ruBuckets, LabelNames = Impl.groupLabels)
-        create ruCfg
-    let latency stat desc = sHistogram (Impl.baseName stat + "_seconds") (Impl.baseDesc desc + " latency")
-    let charge stat desc = ruHistogram (Impl.baseName stat + "_ru") (Impl.baseDesc desc + " charge")
+    let private create (config : Prometheus.HistogramConfiguration) name desc =        
+        let histogram = Prometheus.Metrics.CreateHistogram(name, desc, config)
+        fun tagValues (db, con, group) value ->
+            let labelValues = append tagValues [| db; con; group; |]
+            histogram.WithLabels(labelValues).Observe(value)
+
+    let private create' buckets statSuffix descSuffix (tagNames, tagValues) stat desc =
+        let config = Prometheus.HistogramConfiguration(Buckets = buckets, LabelNames = append tagNames groupLabels)
+        create config (baseName stat + statSuffix) (baseDesc desc + descSuffix) tagValues
+
+    let latencyBuckets = [| 0.0005; 0.001; 0.002; 0.004; 0.008; 0.016; 0.5; 1.; 2.; 4.; 8.; 16. |]
+    let ruBuckets = Prometheus.Histogram.ExponentialBuckets(1., 2., 9)
+
+    let latency = create' latencyBuckets secondsStat latencyDesc
+    let charge = create' ruBuckets rusStat chargeDesc
 
 #if COSMOSSTORE
 open Propulsion.CosmosStore.Log
 #else
 open Propulsion.Cosmos.Log
 #endif
 
-module private Stats =
+/// ILogEventSink that publishes to Prometheus
+type LogSink(tags: string[] * string[]) =
```

Comment:
```suggestion
type LogSink(keys: string[], values : string[]) =
```
I can see some merit in this, but the time to change it was if I'd had you reviewing the first impl in Equinox ;)
arguably the interface should take a `seq<string*string>`,  but if we are changing it, we should do that in Equinox first.

Reply:
We'll get this in Equinox